packages/react-codemirror/src/lang-cypher/autocomplete.ts

```diff
@@ -65,17 +83,70 @@ export const cypherAutocomplete: (config: CypherConfig) => CompletionSource =
       context.explicit,
     );
 
-    return {
-      from: context.matchBefore(/(\w|\$)*$/).from,
-      options: options.map((o) => ({
-        label: o.label,
-        type: completionKindToCodemirrorIcon(o.kind),
-        apply:
-          o.kind === CompletionItemKind.Snippet
-            ? // codemirror requires an empty snippet space to be able to tab out of the completion
-              snippet((o.insertText ?? o.label) + '${}')
-            : undefined,
-        detail: o.detail,
-      })),
-    };
+    if (config.featureFlags.signatureInfoOnAutoCompletions) {
+      return {
+        from: context.matchBefore(/(\w|\$)*$/).from,
+        options: options.map((o) => {
+          let maybeInfo = {};
+          const newDiv = document.createElement('div');
+
+          if (o.signature) {
+            const header = document.createElement('p');
+            header.setAttribute('class', 'cm-completionInfo-signature');
+            header.textContent = o.signature;
+            newDiv.appendChild(header);
+          }
+
+          if (o.documentation) {
+            const paragraph = document.createElement('p');
+            paragraph.textContent = getDocString(o.documentation);
+            newDiv.appendChild(paragraph);
+          }
+
+          maybeInfo = {
+            info: () =>
+              new Promise((resolve) => {
+                resolve(newDiv);
```

Comment:
I think this always attaches the div, even if there's no info. It seems the panel will pop up on all completions. Perhaps it should resolve to `undefined` instead of the div if there's no signature/documentation? 
![CleanShot 2024-05-22 at 13 12 45](https://github.com/neo4j/cypher-language-support/assets/10564538/aae2a7d8-f6e4-4797-bbac-8e5a70280639)

Reply:
Good catch! I've added a few tests to prevent empty boxes like that